text/web-servers/0003-zero-config.md

```diff
@@ -0,0 +1,75 @@
+# Support "zero-configuration" build processes
+
+## Summary
+The NGINX and HTTPD buildpacks within the Web Servers language-family should
+provide a "zero-configuration" option whereby they generate a very simple
+configuration so that the buildpack user does not need to.
+
+## Motivation
+
+Many users have very simple static file deployments. In these cases, requiring
+them to include a bunch of NGINX or HTTPD configuration boilerplate alongside
+their static files can be a burden. Instead, we should allow them to indicate
+that they have some files to serve in the simplest possible way.
+
+## Detailed Explanation
+
+The NGINX and HTTPS buildpacks will implement a common interface through an
+environment variable that will allow them to detect and build when a
+configuration file is not present in the application workspace. Specifically,
+when the `BP_WEB_SERVER` environment variable is set to either `nginx` or
+`httpd` then the respective buildpack will detect and ensure that a simple
+configuration file is included in the built image such that it can serve very
+basic static file applications.
+
+The configuration files will make assumptions about certain options, like the
+location of the files to be served. However, in some cases, the buildpacks
+could choose to allow buildpack users to configure some commonly-specified
+options (e.g. `BP_WEB_SERVER_ROOT=/workspace/my-build-directory`). This RFC
```

Comment:
I think it might be worth wading into the details here. Are we going to take any opinions about the "default" settings? Which options are we willing to support this way vs which options will we require users to byo config file?

Reply:
I'd probably be satisfied with some more specificity around what the process will be for deciding which features to initially include / how to add new configuration features. Just so project maintainers have something to cite when, inevitably, feature requests come in that verge into re-implementing nginx or httpd configuration features.

Reply:
I've detailed those options that it seems reasonable to support. I'd think anything outside of this core group starts to get a bit weedsy and becomes a situation where users really should be using their own configuration file.